pulser/simulation/simulation.py

```diff
@@ -700,11 +781,56 @@ def build_coeffs_ops(basis: str, addr: str) -> list[list]:
             self.operators[addr][basis] = operators
             return terms
 
-        # Time independent term:
+        def build_masked_coeff() -> np.ndarray:
+            coeff = np.zeros(self._tot_duration)
+            try:
+                ti = self._seq._slm_mask_time[0]
+                tf = self._seq._slm_mask_time[1]
+            except IndexError:
+                return coeff
+            coeff[ti:tf] = 1
+            return coeff
+
+        def build_unmasked_coeff() -> np.ndarray:
+            coeff = np.ones(self._tot_duration)
+            try:
+                ti = self._seq._slm_mask_time[0]
+                tf = self._seq._slm_mask_time[1]
+            except IndexError:
+                return coeff
+            coeff[ti:tf] = 0
+            return coeff
+
+        # Interaction term:
         if self.basis_name == "digital" or self._size == 1:
             qobj_list = [0 * self.build_operator([("I", "global")])]
         else:
-            qobj_list = [make_interaction_term()] if self._size > 1 else []
+            if self._seq._slm_mask_targets:
+                um_coeff = build_unmasked_coeff()
+                m_coeff = build_masked_coeff()
+                qobj_list = (
+                    [
+                        [
+                            make_interaction_term(),
+                            self._adapt_to_sampling_rate(um_coeff),
+                        ]
+                    ]
+                    if self._size > 1
+                    else []
+                )
+
+                qobj_list += (
+                    [
+                        [
+                            make_masked_interaction_term(),
+                            self._adapt_to_sampling_rate(m_coeff),
+                        ]
+                    ]
+                    if self._size > 1
+                    else []
+                )
```

Comment:
Bear in mind that `self._size` will be `>1` because of the first `if`.

Reply:
It will? I don't think I ever check if there is more than one qubit when I set the mask (which might actually be a good idea)